Zend/zend_execute.c

```diff
@@ -1896,8 +1896,12 @@ static zend_always_inline void zend_fetch_dimension_address_read(zval *result, z
 			}
 		}
 	} else {
-		if (type != BP_VAR_IS && UNEXPECTED(Z_TYPE_P(container) == IS_UNDEF)) {
-			zval_undefined_cv(EG(current_execute_data)->opline->op1.var, EG(current_execute_data));
+		if (type != BP_VAR_IS) {
+			if (UNEXPECTED(Z_TYPE_P(container) == IS_UNDEF)) {
+				zval_undefined_cv(EG(current_execute_data)->opline->op1.var, EG(current_execute_data));
+			} else if (EXPECTED((EG(current_execute_data)->opline->opcode == ZEND_FETCH_LIST && Z_LVAL_P(dim) == 0 && Z_TYPE_P(container) != IS_FALSE && (Z_TYPE_P(container) != IS_STRING || (Z_TYPE_P(container) == IS_STRING && Z_STRLEN_P(container) != 0))) || (EG(current_execute_data)->opline->opcode != ZEND_FETCH_LIST && (EG(current_execute_data)->opline->op1_type != IS_VAR || (Z_TYPE_P(container) != IS_UNDEF && Z_TYPE_P(container) != IS_NULL && (support_strings && Z_TYPE_P(container) != IS_STRING)))))) {
```

Comment:
I wouldn't be opposed to that.  Might want to have list() continue to exist as is without modification